packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -528,18 +659,153 @@ abstract class SliderComponentShape {
+///    slider of the visual details of the its track.
+///  * [SliderTrackShape] Base component for creating other custom track
+///    shapes.
+class RectangularSliderTrackShape extends SliderTrackShape {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const RectangularSliderTrackShape();
+
+  @override
+  Rect getPreferredRect({
+    RenderBox parentBox,
+    Offset offset = Offset.zero,
+    SliderThemeData sliderTheme,
+    bool isEnabled,
+    bool isDiscrete,
+  }) {
+    final double overlayWidth = sliderTheme.overlayShape.getPreferredSize(isEnabled, isDiscrete).width;
+    final double trackLeft = offset.dx + overlayWidth / 2;
+    final double trackTop = offset.dy + (parentBox.size.height - sliderTheme.trackHeight) / 2;
+    final double trackWidth = parentBox.size.width - overlayWidth;
+    final double trackHeight = sliderTheme.trackHeight;
+    return Rect.fromLTWH(trackLeft, trackTop, trackWidth, trackHeight);
+  }
+
+  // Spacing for disabled slider state.
+  static const double _thumbGap = 2.0;
+
+  @override
   void paint(
     PaintingContext context,
-    Offset thumbCenter, {
-    Animation<double> activationAnimation,
+    Offset offset, {
+    RenderBox parentBox,
+    SliderThemeData sliderTheme,
     Animation<double> enableAnimation,
+    List<double> trackDividers,
+    List<bool> trackSegmentsActive,
     bool isDiscrete,
-    TextPainter labelPainter,
-    RenderBox parentBox,
+    bool isEnabled,
+  }) {
+    assert(trackDividers.isNotEmpty);
+
+    final Rect trackRect = getPreferredRect(
+      parentBox: parentBox,
+      offset: offset,
+      sliderTheme: sliderTheme,
+      isEnabled: isEnabled,
+      isDiscrete: isDiscrete
+    );
+
+    // Used to create a gap around the thumb iff the slider is disabled.
+    double horizontalAdjustment = 0.0;
+    if (!isEnabled) {
+      final double thumbRadius = sliderTheme.thumbShape.getPreferredSize(isEnabled, isDiscrete).width / 2.0;
+      final double gap = _thumbGap * (1.0 - enableAnimation.value);
+      horizontalAdjustment = thumbRadius + gap;
+    }
+
+    final ColorTween activeTrackColorTween = ColorTween(begin: sliderTheme.disabledActiveTrackColor , end: sliderTheme.activeTrackColor);
+    final ColorTween inactiveTrackColorTween = ColorTween(begin: sliderTheme.disabledInactiveTrackColor , end: sliderTheme.inactiveTrackColor);
+    final Paint activePaint = Paint()..color = activeTrackColorTween.evaluate(enableAnimation);
+    final Paint inactivePaint = Paint()..color = inactiveTrackColorTween.evaluate(enableAnimation);
+
+    final List<double> allTrackDividers = <double>[]..add(0.0)..addAll(trackDividers)..add(1.0);
+    double trackSegmentLeft = trackRect.left + allTrackDividers[0] * trackRect.width;
+    double trackSegmentRight;
+    for (int i = 1; i < allTrackDividers.length; i++) {
```

Comment:
I might be reading this wrong/not understanding, but won't this skip the first and last dividers?

Reply:
The first and last dividers are always 0.0 and 1.0, which are added to allTrackDividers. This way, the track paint API only needs the middle dividers

Reply:
ack